tests/src/test/scala/com/nvidia/spark/rapids/RegularExpressionParserSuite.scala

```diff
@@ -0,0 +1,73 @@
+/*
+ * Copyright (c) 2021, NVIDIA CORPORATION.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package com.nvidia.spark.rapids
+
+import scala.collection.mutable.ListBuffer
+
+import org.scalatest.FunSuite
+
+import org.apache.spark.sql.rapids.{QuantifierFixedLength, RegexAST, RegexChar, RegexCharacterClass, RegexCharacterRange, RegexChoice, RegexGroup, RegexParser, RegexRepetition, RegexSequence, SimpleQuantifier}
+
+class RegularExpressionParserSuite extends FunSuite {
+
+  test("simple quantifier") {
+    assert(parse("a{1}") ===
+      RegexSequence(ListBuffer(
+      RegexRepetition(RegexChar('a'), QuantifierFixedLength(1)))))
+  }
+
+  test("not a quantifier") {
+    assert(parse("{1}") ===
+      RegexSequence(ListBuffer(
+        RegexChar('{'), RegexChar('1'),RegexChar('}'))))
+  }
+
+  test("nested repetition") {
+    assert(parse("a*+") ===
+      RegexSequence(ListBuffer(
+        RegexRepetition(
+          RegexRepetition(RegexChar('a'), SimpleQuantifier('*')),
+            SimpleQuantifier('+')))))
+  }
+
+  test("choice") {
+    assert(parse("a|b") ===
+      RegexChoice(RegexSequence(ListBuffer(RegexChar('a'))),
+        RegexSequence(ListBuffer(RegexChar('b')))))
+  }
+
+  test("group") {
+      assert(parse("(a)(b)") ===
+        RegexSequence(ListBuffer(
+          RegexGroup(RegexSequence(ListBuffer(RegexChar('a')))),
+          RegexGroup(RegexSequence(ListBuffer(RegexChar('b')))))))
+  }
+
+  test("character class") {
```

Comment:
Are there more tests planned, such as testing hex digit parsing, octal parsing, etc.?

Reply:
I've added tests for hex and octal. Most of the testing is currently happening via the transpiler suite but I think it would be good to have more pure parser tests so I will keep plugging away at this and expand the tests here.